### PR TITLE
Log trace data for discarded traces

### LINKED ISSUE #2776: How to get trace names of too big traces?
When spans are failing because of their huge size, I don't know how can I check which spans are that, as it is impossible to retrieve them from Tempo. 

Tempo-compactor logs:
```
compactor.go:248 msg="max size of trace exceeded" tenant=single-tenant traceId=<trace-id> discarded_span_count=<count>
```
When I'm using the traceID to look for span in tempo, I'm getting the response: 
```
failed to get trace with id: <id> Status: 500 Internal Server Error Body: response larger than the max (<size> vs <size>)
```

It would be great if tempo-compactor could log the span name.  Or, is there other way to get the span information, without tweaking my settings? 

Thanks a lot for your time and sorry if it's newbie question.

### PR DESCRIPTION
**Pull request by MarcinGinszt**
**What this PR does**:
This PR adds the RootSpanName and RootServiceName to log about discarded span. 
I added them to main log notifying about this event, since those are basic data allowing identification of the span, and this log doesn't occur often (or, atleast, it should not do so :slightly_smiling_face:)

I decided to not include test update at least in the first version of PR, as it looks like they would demand significant changes. 
In order to simulate the rootSpanName and rootSpanService, we would need to change the MakeBatch function preparing the spans for tests- first span `ParentSpanId` [would have to be empty](https://github.com/grafana/tempo/commit/b4e11d896c8bc06bc9b39e13f1e9d46723d349c5#diff-ffff7cdedcf55972b0441421e0be7cd1c21724e7469d0ce57659a24aaf7b6b08R141). However, definition of "empty" is different between different encodings (empty byte array or simply nil). 
As far as I understood, because of this issue testing this feature would demand replicating huge chunks of testing library for different encodings, or additional logic changes. Otherwise, multiple tests from different packages would expect `nil` and would get `[]byte{}` as a value of first span ParentSpanID- or vice versa. 

**Which issue(s) this PR fixes**:
Fixes #2776

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by joe-elliott — COMMENTED** [review]
> I decided to not include test update at least in the first version of PR,

Can you extend this test to make sure it extracts the fields correctly?
https://github.com/grafana/tempo/blob/main/tempodb/encoding/vparquet3/compactor_test.go#L150

I think that would be worthwhile. I completely agree it's not worth writing a test to make sure a log line gets written as expected.

Thanks for taking a look at this!

File: tempodb/encoding/vparquet3/compactor.go [hunk 1]
```diff
@@ -321,15 +321,25 @@ func estimateMarshalledSizeFromParquetRow(row parquet.Row) (size int) {
 // countSpans counts the number of spans in the given trace in deconstructed
 // parquet row format and returns traceId.
 // It simply counts the number of values for span ID, which is always present.
-func countSpans(schema *parquet.Schema, row parquet.Row) (traceID string, spans int) {
+func countSpans(schema *parquet.Schema, row parquet.Row) (traceID string, rootSpanName string, rootServiceName string, spans int) {
```
[the remark this answers is not in the document]
Comment (MarcinGinszt) [reply tempodb/encoding/vparquet3/compactor.go:324]: `countSpans` return values are argument to the same interface- [SpansDiscarded](https://github.com/grafana/tempo/pull/2816/files#diff-cf86b7346e8f2d12330637bff690d0fca5d738e9b37ee0fd96cd4e348210b631R78) for all the Parquet versions. 
Should I return empty strings as rootSpanName and rootServiceName  in the v1 and v2?

**Comment by MarcinGinszt** [comment]
@joe-elliott 
>Can you extend this test to make sure it extracts the fields correctly?
> https://github.com/grafana/tempo/blob/main/tempodb/encoding/vparquet3/compactor_test.go#L150
> 
> I think that would be worthwhile. I completely agree it's not worth writing a test to make sure a log line gets written as expected.

I'm sorry, I'm afraid I don't know how to do that without replicating huge parts of `pkg/util/test/req.go`, so that batch is created differently for v1 and v3. I wrote detailed description of the problem in the PR description. 
Should I proceed with replicating `make batch` for different versions?

File: tempodb/encoding/vparquet3/compactor.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (joe-elliott) [reply tempodb/encoding/vparquet3/compactor.go:324]: honestly, it's not a big deal either way, and you've already gone through the trouble of writing the code in each version. I'd say just leave it as is.

**Comment by joe-elliott** [comment]
> Should I proceed with replicating MakeBatch for different versions?

Nah, we don't have to do anything complex here.

> I'm sorry, I'm afraid I don't know how to do that without replicating huge parts of pkg/util/test/req.go

Is something like this not sufficient:

image

File: tempodb/encoding/vparquet3/block_findtracebyid.go [hunk 2]
```diff
@@ -24,7 +24,9 @@ const (
 	SearchNext     = -2
 	NotFound       = -3
 
-	TraceIDColumnName = "TraceID"
+	TraceIDColumnName         = "TraceID"
+	RootSpanNameColumnName    = "RootSpanName"
```
[the remark this answers is not in the document]
Comment (MarcinGinszt) [reply tempodb/encoding/vparquet3/block_findtracebyid.go]: [Applied](https://github.com/grafana/tempo/pull/2816/commits/c0a524601cfb439f9f3cf826c7dc1531cf93f5ef) together with tests update- thanks for hint regarding them :)

**Review by joe-elliott — APPROVED** [review]
Nice addition. Thanks!
